platform/ios/src/MGLAnnotationView.mm

```diff
@@ -157,6 +161,26 @@ - (void)updateScaleTransformForViewingDistance
     }
 }
 
+- (void)setRotatesToMatchCamera:(BOOL)rotatesToMatchCamera
+{
+    if (_rotatesToMatchCamera != rotatesToMatchCamera)
+    {
+        _rotatesToMatchCamera = rotatesToMatchCamera;
+        [self updateRotateTransform];
+    }
+}
+
+- (void)updateRotateTransform
+{
+    if (self.rotatesToMatchCamera == NO) return;
+
+    CGFloat directionRad = self.mapView.direction * M_PI / 180.0;
+    CATransform3D newRotateTransform = CATransform3DMakeRotation(-directionRad, 0, 0, 1);
+    self.layer.transform = CATransform3DConcat(CATransform3DIdentity, newRotateTransform);
```

Comment:
@boundsj, is it correct to start with `CATransform3DIdentity` here, or should we start with `self.layer.transform` and `undoOfLastScaleTransform` as above in `-updateScaleTransformForViewingDistance`?